### ISSUE TITLE
After Installing and uninstalling vmware workstation 15.5 WSL is stopped working

### ISSUE THREAD

**Issue by 3axap4eHko** [microsoft/WSL#5367]
# Environment

```none
Windows build number: Microsoft Windows NT 10.0.19041.0
Your Distribution version: N/A
Whether the issue is on WSL 2 and/or WSL 1: WSL 2
```

# Steps to reproduce

1. Enable WSL 2
2. Restart
3. Install WMware Workstation 15.5
4. Restart
5. Uninstall WMware Workstation 15.5
6. Restart
7. Run any WSL command `wsl -l -v` for example

#  Expected behavior

Should execute the command

# Actual behavior

Nothing is happening. Powershell hangs out for 5 mins and throws an error

![image](https://user-images.githubusercontent.com/1630657/84070824-bbdb7180-a99a-11ea-82f3-1bab238d0841.png)

# Already tried

1. Disable, Restart and Enable again all virtualization related features
2. Re-Install WSL2 kernel
3. Install the latest updates

BTW. Hyper-V also does not work anymore

# Would like to know

1. How to debug such cases?
2. Where is WSL logs (not the Linux distributive logs)?
3. How to completely reinstall WSL?

**Comment by jonathanpallant** [comment]
I see this too after uninstalling the VMware 16 beta.

**Comment by yadalik** [comment]
Maybe this will help:

https://weblogs.asp.net/dixin/run-hyper-v-and-vmware-virtual-machines-on-windows-10

tl;dr:

`bcdedit /set hypervisorlaunchtype off` - to enable vmware things, disables wsl

`bcdedit /set hypervisorlaunchtype auto` - to enable hyper-v (wsl) things, disables vmware

BTW, there is beta version of vmware, that should work in wsl environment, but only starting from Haswell CPUs, see link here, notice - it destroyed someones windows installation, so be careful:

1. https://blogs.vmware.com/workstation/2020/01/vmware-workstation-tech-preview-20h1.html

2. https://docs.microsoft.com/en-us/windows/wsl/wsl2-faq#will-i-be-able-to-run-wsl-2-and-other-3rd-party-virtualization-tools-such-as-vmware-or-virtualbox

**Comment by 3axap4eHko** [comment]
@yadalik nothing helps. After installing fresh windows works again. I'm using VMWare beta
